src/lib/pickles/pickles.ml

```diff
@@ -3062,16 +3059,15 @@ module Make_str (_ : Wire_types.Concrete) = struct
             let x =
               exists Field.typ ~compute:(fun () -> Field.Constant.of_int 3)
             in
-            let y =
-              exists Field.typ ~compute:(fun () -> Field.Constant.of_int 0)
-            in
-            let z =
-              exists Field.typ ~compute:(fun () -> Field.Constant.of_int 0)
-            in
+            (* y and z share the same [compute] function *)
+            let compute () = Field.Constant.zero in
```

Comment:
I feel like this hurts readability rather than helping it.. Revert?

Reply:
I could agree that this does not belong to this PR. 

But I disagree that it hurts readability (I even find it more readable)  *and* it still does 1 less allocation than the original code.

Reply:
Removed for the first reason

Reply:
The way we currently use snarky, it almost never makes sense to factor out the `compute` argument. Conceptually, it's the same as an if statement:
```
let foo =
  if running_as_prover then (* ... *)
  else (* implicit dummy value *)
in
```
which translates to
```
let foo =
  exists typ ~compute:(fun () -> (* ... *))
in
```

IMO, this change obscures the control flow. I still prefer the original.